portalnet/src/discovery.rs

```diff
@@ -111,28 +111,33 @@ impl Discovery {
                 .map_err(|e| format!("When adding key to servers ENR: {e:?}"))?
         };
 
-        // Check if we have an old version of our Enr and if we do, increase our sequence number
-        if let Some(enr_file_location) = portal_config.enr_file_location {
-            let trin_enr_file_location = enr_file_location.join(ENR_FILE_NAME);
+        // This should only never run in test cases.
```

Comment:
`// This should never run in test cases.` - much less confusing in my opinion. But also the fact that this check needs to happen feels off. If we go with my suggestion about `node_data_dir` then we can just get rid of this check completely

Reply:
The issue I have with this is some Discovery tests don't use a path, but I cause I can just generate a tmp path for those